go/contacts/contacts.go

```diff
@@ -169,6 +220,10 @@ func ResolveContacts(mctx libkb.MetaContext, provider ContactsProvider, contacts
 	// contact name and hold the same assertion. Will also skip same assertions
 	// within one contact (duplicated components with same value and same or
 	// different name)
+	type contactAssertionPair struct {
```

Comment:
defining type inside a function?

Reply:
maybe pass c.Name into AssertionFromComponent and return this type from there

Reply:
I don't think there's a need to. This type is only used here as key to a map for deduplicating `(contactName,assertionValue)` values.